component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/listener/OauthDPoPInterceptorHandlerProxy.java

```diff
@@ -0,0 +1,318 @@
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.core.handler.AbstractIdentityHandler;
+import org.wso2.carbon.identity.core.model.IdentityEventListenerConfig;
+import org.wso2.carbon.identity.core.util.IdentityConfigParser;
+import org.wso2.carbon.identity.core.util.IdentityCoreConstants;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.dpop.util.DPoPConstants;
+import org.wso2.carbon.identity.dpop.util.OuthTokenType;
+import org.wso2.carbon.identity.oauth.common.DPoPTokenState;
+import org.wso2.carbon.identity.oauth.common.exception.InvalidOAuthClientException;
+import org.wso2.carbon.identity.oauth.dao.OAuthAppDO;
+import org.wso2.carbon.identity.oauth.event.AbstractOAuthEventInterceptor;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2AccessTokenReqDTO;
+import org.wso2.carbon.identity.oauth2.model.HttpRequestHeader;
+import org.wso2.carbon.identity.oauth2.token.OAuthTokenReqMessageContext;
+import org.wso2.carbon.identity.oauth2.token.bindings.TokenBinding;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+
+import javax.ws.rs.HttpMethod;
+import javax.xml.namespace.QName;
+import java.security.interfaces.ECPublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.sql.*;
+import java.text.ParseException;
+import java.util.Date;
+import java.util.Map;
+
+import static org.wso2.carbon.identity.dpop.util.DPoPConstants.INVALID_DPOP_ERROR;
+
+/**
+ * This class extends AbstractOAuthEventInterceptor and listen to oauth token related events. In this class, DPoP proof validation
+ * will be handle for DPoP token requests
+ */
+public class OauthDPoPInterceptorHandlerProxy extends AbstractOAuthEventInterceptor {
+
+    private static final Log log = LogFactory.getLog(OauthDPoPInterceptorHandlerProxy.class);
+    private int dPopValidity;
+    private static final String ECDSA_ENCRYPTION = "EC";
+    private static final String RSA_ENCRYPTION = "RSA";
+    private boolean isRefreshRequest = false;
+
+    /**
+     * This method handles dpop proof validation during pre token issuance.
+     *
+     * @param tokenReqDTO  OAuth2AccessTokenReqDTO.
+     * @param tokReqMsgCtx OAuthTokenReqMessageContext.
+     * @param params       Map of parameters.
+     * @throws IdentityOAuth2Exception
+     */
+    @Override
+    public void onPreTokenIssue(OAuth2AccessTokenReqDTO tokenReqDTO, OAuthTokenReqMessageContext tokReqMsgCtx,
+                                Map<String, Object> params) throws IdentityOAuth2Exception {
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("Listening to the pre token issue event with the DPoP proof for the " +
+                    "application: %s", tokenReqDTO.getClientId()));
+        }
+        isRefreshRequest = false;
```

Comment:
Lets pass this in as a parameter to the required method rather that keeping as a class variable